### PR TITLE
Remove the built-in "sustain" watchdog:

### PR DESCRIPTION
**Pull request by nbougalis**
The built-in watchdog is simplistic and can, sometimes, cause problems especially on systems that have the ability to automatically start and monitor processes.

This commit removes the sustain system entirely, changes the handling of the `SIGTERM` signal to properly terminate the process and improves the error message reported to the user when the command line used to start `rippled` is incorrect and malformed.

### CODE REVIEW

**Comment by MarkusTeufelberger** [comment]
Thanks a lot, this makes it much easier to monitor `rippled` in production as one can actually see if a server crash-loops without having to query its uptime constantly via API calls! :-)

File: src/ripple/app/main/Main.cpp [hunk 1]
```diff
@@ -403,6 +401,8 @@ int run (int argc, char** argv)
         "so this option is not needed for users")
     ("unittest-child",
         "For internal use only when spawning child unit test processes.")
+    ("fg",
+        "Depreceated: server always in foreground mode.")
```
Comment (ximinez) [inline src/ripple/app/main/Main.cpp:405]: "Deprecated"
* https://www.dictionary.com/browse/deprecate
* https://en.wikipedia.org/wiki/Deprecation

Comment (nbougalis) [reply src/ripple/app/main/Main.cpp:405]: D’oh! I know better than that.

**Comment by nbougalis** [comment]
If merged, this PR also addresses #2822, which has been closed for a while.

File: Builds/containers/shared/rippled.service [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+After=network-online.target
+Wants=network-online.target
 
 [Service]
 Type=simple
 ExecStart=/opt/ripple/bin/rippled --net --silent --conf /etc/opt/ripple/rippled.cfg
-# Default KillSignal can be used if/when rippled handles SIGTERM
-KillSignal=SIGINT
-Restart=no
+Restart=yes
```
Comment (carlhua) [inline Builds/containers/shared/rippled.service:9]: What is the intention of this line? Based on the documentation of systemd.service, it did not list =yes as an option. There are various configuration we can pass to Restart= depending on what we want. do we want to restart the service even if it has a clean exit code initiated by an operator? I used this link as my reference http://manpages.ubuntu.com/manpages/cosmic/man5/systemd.service.5.html

Comment (nbougalis) [reply Builds/containers/shared/rippled.service:9]: Yeah, sorry. Brain shortcircuit, this should be `on-failure`

File: src/ripple/app/main/Application.cpp [hunk 3]
```diff
@@ -1296,10 +1276,33 @@ class ApplicationImp
 bool ApplicationImp::setup()
 {
     // We want to intercept and wait for CTRL-C to terminate the process
-    m_signals.add (SIGINT);
+    m_signals.add(SIGINT);
+    m_signals.add(SIGTERM);
+    m_signals.add(SIGHUP);
```
[the remark this answers is not in the document]
Comment (nbougalis) [reply src/ripple/app/main/Application.cpp:1281]: Ahh, good ole' Windows. I'll fix this.

**Comment by nbougalis** [comment]
This PR has been cherry-picked into into #3326. It will be merged together with that PR.

**Pull request closed without merging**